apps/remote_control/lib/lexical/remote_control/build/error.ex

```diff
@@ -12,19 +12,80 @@ defmodule Lexical.RemoteControl.Build.Error do
     detail_diagnostics = detail_diagnostics(detail)
     error = message_info_to_binary(message_info, token)
     error_diagnostics = parse_error_to_diagnostics(context, error, token)
-    error_diagnostics ++ detail_diagnostics
+    extract_line(error_diagnostics ++ detail_diagnostics)
   end
 
   def parse_error_to_diagnostics(context, message_info, token) do
-    diagnostic = %Diagnostic{
-      file: nil,
-      severity: :error,
-      position: context_to_position(context),
-      compiler_name: "Elixir",
-      message: message_info_to_binary(message_info, token)
-    }
+    parse_error_diagnostic_functions = [
+      &build_end_line_diagnostics/3,
+      &build_start_line_diagnostics/3,
+      &build_hint_diagnostics/3
+    ]
+
+    Enum.flat_map(parse_error_diagnostic_functions, & &1.(context, message_info, token))
+  end
+
+  defp extract_line(diagnostics) do
+    # We need to uniq by position because the same position can be reported
+    # and the `end_line_diagnostic` is always the precise one
+    position_key = fn
```

Comment:
nit: name this `extract_line`
Also, don't mark things resolved if they're not resolved, it wastes time.

Reply:
Sorry, I misunderstood the number of lines you referred to earlier. I changed, and I think It's all good now.